runtime/src/main/java/org/corfudb/runtime/collections/StreamManager.java

```diff
@@ -257,6 +258,10 @@ public void run() {
                     if (logData == null) {
                         break; // Stream is all caught up, sleep for a bit.
                     }
+                    // Avoid LogData decompression/deserialization if no table of interest contained.
+                    if (Sets.intersection(logData.getStreams(), tablesOfInterest.keySet()).isEmpty()) {
+                        continue;
```

Comment:
The transaction stream never appears in tablesOfInterest, is that right?

Reply:
Yes, it should never appear, tableOfInterest is user specified, they don't know about the transaction stream.